src/datapoints-graph/alarm-event-selector/pipes/includes-alarm.pipe.ts

```diff
@@ -0,0 +1,25 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import {
+  AlarmDetails,
+  AlarmOrEvent,
+  EventDetails,
+} from '../alarm-event-selector.model';
+
+@Pipe({
+  name: 'includesAlarmOrEvent',
+})
+export class IncludesAlarmOrEventPipe implements PipeTransform {
+  transform(itemList: EventDetails[], item?: EventDetails): boolean;
+  transform(itemList: AlarmDetails[], item?: AlarmDetails): boolean;
+  transform(itemList: any[], item?: any): boolean {
```

Comment:
Any thoughts on this approach?
```suggestion
  transform<T extends AlarmDetails | EventDetails>(itemList: T[], item?: T): boolean {
```